### PR TITLE
London_8_Delroy_Gayle_JS1_WK2

### PR DESCRIPTION
**Pull request by DelroyGayle**
London_8_Delroy_Gayle_JS1_WK2
-->

- What do you still not understand?

I don't always find the instructions to the task clear. For example, 1-radio-stations.js

I was not altogether clear what was being asked for.

### CODE REVIEW

File: exercises/G-conditionals-2/exercise-3.js [hunk 1]
```diff
@@ -8,7 +8,18 @@
...
+      return 'A'
+  else
+  if (mark > 60) 
+      return 'B'
+  else
+  if (mark >= 50) 
+      return 'C'
+  else
+      return 'F';
+ }
```
Comment (hachi-ops) [inline exercises/G-conditionals-2/exercise-3.js:22]: I actually didn't have a clue that you can skip internal curly brackets within if/else if statements and the function would still work. Thanks for enlightening me:)

File: exercises/G-conditionals-2/exercise-4.js [hunk 2]
```diff
@@ -8,7 +8,9 @@
   Hint: Google how to check if a string contains a word
 */
 
-function containsCode(sentence) {}
+function containsCode(sentence) {
+    return ( sentence.search("code") >= 0)
```
Comment (hachi-ops) [inline exercises/G-conditionals-2/exercise-4.js:12]: Awesome. I used : (sentence.includes("code")) which is just another way of doing the same thing. I will try to remember yours.

File: exercises/I-array-properties/exercise.js [hunk 3]
```diff
@@ -6,7 +6,7 @@
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  return arr.length === 0; // complete this statement
```
Comment (hachi-ops) [inline exercises/I-array-properties/exercise.js:9]: Nice.

File: exercises/J-array-get-set/exercise.js [hunk 4]
```diff
@@ -5,11 +5,11 @@
 */
 
 function first(arr) {
-  return; // complete this statement
+  return arr[0]; // complete this statement
```
Comment (hachi-ops) [inline exercises/J-array-get-set/exercise.js:8]: Awesome.

File: exercises/J-array-get-set/exercise.js [hunk 5]
```diff
@@ -5,11 +5,11 @@
...
 }
 
 function last(arr) {
-  return; // complete this statement
+  return arr[arr.length-1]; // complete this statement
```
Comment (hachi-ops) [inline exercises/J-array-get-set/exercise.js:12]: Awesome.

**Review by hachi-ops — COMMENTED** [review]
Hi Delroy, I reviewed your exercises folder. I hope you'll find my suggestions helpful/relevant.

File: exercises/D-logical-operators/exercise.js [hunk 6]
```diff
@@ -11,14 +11,14 @@ let cssLevel = 4;
...
-let cssLevelAbove5;
+let htmlLevelAbove5 = htmlLevel > 5;
+let cssLevelAbove5 = cssLevel > 5;
 
 // Finish the next two statement
 // Use the previous variables and logical operators
 // Do not "hardcode" the answers
-let cssAndHtmlAbove5;
-let cssOrHtmlAbove5;
+let cssAndHtmlAbove5 = cssLevelAbove5 && htmlLevelAbove5;
```
Comment (amellie) [inline exercises/D-logical-operators/exercise.js:20]: perfect!

File: exercises/J-array-get-set/exercises2.js [hunk 7]
```diff
@@ -7,6 +7,8 @@
 */
 
 let numbers = [1, 2, 3]; // Don't change this array literal declaration
+numbers[3] = 4; // OR numbers[numbers.length] = 4;
```
Comment (amellie) [inline exercises/J-array-get-set/exercises2.js:10]: you can also use the ``push`` function to achieve this

File: mandatory/1-fix-functions.js [hunk 8]
```diff
@@ -32,7 +36,7 @@ function greaterThan10(num) {
 
 function get3rdIndex(arr) {
```
Comment (amellie) [inline mandatory/1-fix-functions.js:37]: clean!

File: mandatory/2-function-creation.js [hunk 9]
```diff
@@ -16,7 +18,19 @@ Write a function that:
 - numbers greater 100 must be replaced with 100
 */
 
-function formatPercentage(num) {}
+function formatPercentage(num) {
+      if (num >= 100)
+          return "100%";
+      let str = num.toFixed(2);
+
+      if (str.slice(-2) === "00")
```
Comment (amellie) [inline mandatory/2-function-creation.js:26]: I would use ``parseFloat`` which simplifies the whole thing

File: mandatory/2-function-creation.js [hunk 10]
```diff
@@ -25,7 +39,35 @@ Write a function that:
 - removes any forward slashes (/) in each string
 - makes all strings all lowercase
 */
-function tidyUpStrings(arrayOfStrings) {}
+function tidyUpStrings(arrayOfStrings) {
+  let pos;
+  for (let i = 0; i < arrayOfStrings.length; i++) {
+        let aString = arrayOfStrings[i].trim(); // removes any spaces in the beginning or end of each string
+        while ((pos = aString.indexOf("/")) >= 0) {
```
Comment (amellie) [inline mandatory/2-function-creation.js:46]: there is a ``replace()`` function that you can use so you shouldn't be doing all this while loop

File: mandatory/3-playing-computer.js [hunk 11]
```diff
@@ -7,12 +7,33 @@
...
+
+  Ans: 2, 6, 4, 9, 6, 13, 8
+
   4. How many times is "f1" called?
+
+  Ans: 2 times
+
   5. How many times is "f2" called?
+
+  Ans: 2 times
```
Comment (amellie) [inline mandatory/3-playing-computer.js:27]: incorrect, check again :)

File: mandatory/3-playing-computer.js [hunk 12]
```diff
@@ -7,12 +7,33 @@
...
+
   6. What value does the "a" parameter take in the first "f1" call?
+
+  Ans: 8
```
Comment (amellie) [inline mandatory/3-playing-computer.js:31]: not quite...

File: mandatory/3-playing-computer.js [hunk 13]
```diff
@@ -7,12 +7,33 @@
...
+
   7. What is the value of the "a" outer variable when "f1" is called for the first time?
+
+  Ans: 8
```
Comment (amellie) [inline mandatory/3-playing-computer.js:35]: yes correct

File: extra/1-radio-stations.js [hunk 14]
```diff
@@ -15,6 +15,15 @@
 
 // `getAllFrequencies` goes here
 
+function getAllFrequencies() {
```
Comment (amellie) [inline extra/1-radio-stations.js:18]: perfect!

**Review by amellie — COMMENTED** [review]
Hi Delroy, a lot of the comments have been added by the previous reviewer, so these are the additional ones. A lot of them are minor, except for one or two. I can see you are doing really well with JS. Well done!

File: mandatory/3-playing-computer.js (hunk 11 above)
Comment (DelroyGayle) [reply mandatory/3-playing-computer.js:27]: 3 times

File: mandatory/3-playing-computer.js (hunk 12 above)
Comment (DelroyGayle) [reply mandatory/3-playing-computer.js:31]: 1

**Pull request closed without merging**